stack.hpp

```diff
@@ -7,6 +7,11 @@ class Stack {
 public:
     Stack(std::size_t capacity): storage_(new T[capacity]), capacity_(capacity) {}
 
+    ~Stack(){
+        delete[] storage_;
+        storage_ = nullptr;
```

Comment:
We could argue if this is actually needed...

The object is about to be destroyed, so this is as if you would clean a room in a house that is about to burn down anyway...